ffi/diplomat/src/properties_iter.rs

```diff
@@ -0,0 +1,40 @@
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use core::ops::RangeInclusive;
+
+    /// Result of a single iteration of [`CodePointRangeIterator`].
+    /// Logically can be considered to be an `Option<RangeInclusive<u32>>`,
+    ///
+    /// `start` and `end` represent an inclusive range of code points [start, end],
+    /// and `done` will be true if the iterator has already finished. The last contentful
+    /// iteration will NOT produce a range done=true, in other words `start` and `end` are useful
+    /// values if and only if `done=false`.
+    pub struct CodePointRangeIteratorResult {
```

Comment:
I'm not sure if this is the best way to do this

Eventually we will have Options